packages/strapi-design-system/src/Tag/Tag.js

```diff
@@ -25,35 +25,42 @@ const TagText = styled(Typography)`
 export const Tag = ({ children, icon, disabled, onClick, ...props }) => {
   const handleClick = (e) => {
     if (disabled) return;
-    onClick(e);
+    if (onClick) onClick(e);
   };
 
   return (
-    <TagWrapper
-      as="button"
+    <Box
+      display="inline-block"
       background={disabled ? 'neutral200' : 'primary100'}
       color={disabled ? 'neutral700' : 'primary600'}
       paddingLeft={3}
-      paddingRight={3}
-      onClick={handleClick}
-      aria-disabled={disabled}
       borderWidth="1px"
       borderStyle="solid"
       borderColor={disabled ? 'neutral300' : 'primary200'}
       hasRadius
       height={`${32 / 16}rem`}
       {...props}
     >
-      <Flex>
+      <Flex height="100%">
         <TagText disabled={disabled} variant="pi" fontWeight="bold" as="span">
           {children}
         </TagText>
 
-        <Box paddingLeft={2}>
-          <Flex>{icon}</Flex>
-        </Box>
+        <TagIconWrapper
+          as="button"
+          aria-disabled={disabled}
```

Comment:
I'm wondering if we should set the `disabled` attribute instead? Why? Because then the button would never receive any event that need to be canceled in `handleClick()`. I'm not sure if this would harm a11y, but I think if the button is disabled, it shouldn't be possible to focus it at all ... ?

Reply:
Maybe we should have both?

I wanted to opt for `disabled` as well, but the previous implementation of the `Tag` component was using `aria-disabled` to conditionally apply styles in the styled component, so I decided to keep that since the root already had `disabled`.

Reply:
We are not using the disabled attribute because it makes the button unfocusable. When using voice over tools, a user have to now that a button exist and is disabled.

Reply:
@HichamELBSI I agree that for most cases that makes sense, but I'd argue that we don't need the button to be focusable in this case, because the user doesn't need to know it even exists.

Reply:
hmm 🤔, in which case it is not important for the user to know if this or that element of the page exists?

IMO, we exclude some users because the disabled state is visible but the button will be totally unknown for another user (who uses screen reader tools).

Also, what do we know about the context where this component will be used? We know what we do with it on Strapi but this library is used in other products and in Strapi plugins that will implement this button in many other context than just deletion. What do you think about it?

Reply:
After giving this some thought I think you are right, @HichamELBSI. It does matter in this case.